### ISSUE TITLE
[Master feature] Loading state improvements for amp-list

### ISSUE THREAD

**Issue by ericlindley-g** [ampproject/amphtml#14063]
amp-list is associated with multiple loading states, particularly when integrated w/ with amp-bind & infinite scroll. Some of these states can already be configured by developers and get a default loading state in order to communicate status to users. This feature is to ensure that more of these states are configurable by developers and get good default states to the extent possible

Issues:
• Content not loaded (initial page load)
• src changed by amp-bind, content not yet loaded
• content to be appended via button or scroll-trigger, not yet loaded

**Comment by ampprojectbot** [comment]
This issue doesn't have a category which makes it harder for us to keep track of it. @ericlindley-g Please add an appropriate category.

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @aghassemi Do you have any updates?

**Comment by aghassemi** [comment]
dedup with https://github.com/ampproject/amphtml/issues/12212
